Automattic/wp-calypso
# Skip Plans step in Sensei onboarding flow if user is coming from SenseiLMS.com

Related to #85170

## Proposed Changes

* Auto-select billing period in the plans step in Sensei onboarding flow if the user is coming from a URL with the `plan` param in the URL, and it matches one of the options: "MONTHLY" and "ANNUALLY".
* In parallel, I made 2 changes on senseilms.com:
  * Remove the tooltip about monthly plans from senseilms.com, as I commented in https://github.com/Automattic/wp-calypso/issues/85170#issuecomment-1865145282.
  * Add the `plan` argument to the link "Create A Site" from https://senseilms.com/pricing/.

> I do wonder if it would make sense to create the site earlier in the process—maybe after the domain step—and in the background, so it doesn't block other steps and we can reduce wait time and a loading screen. None of the things [createAndConfigureSite](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/create-sensei-site.ts#L66C8-L66C30) does is dependent on user input.

The next step after the domain step, is the plans step. So it is now immediately starting the site creation after the domain step when the user is coming from senseilms.com. I thought to make it in parallel with the the domain step, and I started it in https://github.com/Automattic/wp-calypso/pull/85810/commits/4a97317c1ca69d967d82349dbd996c67d96346c7 but I noticed the site creation [depends on the site domain](https://github.com/Automattic/wp-calypso/blob/2bdb35a42e47eb444336b997d5172adabed6a4f1/packages/data-stores/src/onboard/actions.ts#L195), so I reverted it https://github.com/Automattic/wp-calypso/pull/85810/commits/cdd44c158f202a8be8a5570a53f02f1857271aa1. It seems it could use other inputs as fallback, but if we select a domain, and then try to select another one, it would create a site with the previous domain, so it would demand some other changes with the site creation to avoid issues with edge cases then I decided to skip it for now.

What we could try to do in the future, is to run some things from `SenseiLaunch` in parallel with the purpose step. I didn't dig into that to see if that's totally feasible, but I think so. We probably only need to run the `savePurposeData` later.

## Testing Instructions



* Navigate to `https://senseilms.com/pricing/`.
* Make sure the Sensei Bundle doesn't have a tooltip talking about monthly prices anymore.
* Copy the link from "Create A Site".
* Replace the domain with `http://calypso.localhost:3000`.
* Go through the steps.
* Make sure you're not asked to choose a plan.
* Make sure the annual billing period is selected.
* Make sure the whole process works properly, and the site is created correctly.
* Now navigate to `http://calypso.localhost:3000/setup/sensei/`.
* Go through the steps, and make sure you're asked to choose a plan.
* Make sure the whole process works properly, and the site is created correctly.
* Test again with `http://calypso.localhost:3000/setup/sensei/?plan=ANYTHING`, and make sure the plan is ignored.

## Pre-merge Checklist
